### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by zadziarnouksi-taras**

### CODE REVIEW

File: src/main/java/by/zadziarnouski/workoutlog/configuration/ConfigurationCloudinary.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@Configuration
+public class ConfigurationCloudinary {
+
+    @Bean
+    public Cloudinary ConnectionToCloudinary(){
+        Cloudinary cloud = new Cloudinary(ObjectUtils.asMap(
+                "cloud_name", "edmonddantes",
+                "api_key", "591843375611729",
+                "api_secret", "ZwF2t7X8A0b-yaxzVS8to0zcR3w"));
+        return cloud;
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/configuration/ConfigurationCloudinary.java:17]: such kind of properties better to store in repository variables (if ci/cd flow implemented) or in .properties file

File: src/main/java/by/zadziarnouski/workoutlog/configuration/DatabaseFillingConfig.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+@Configuration
+public class DatabaseFillingConfig {
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/configuration/DatabaseFillingConfig.java:18]: All this data better to store in .sql file and execute it on application start

File: src/main/java/by/zadziarnouski/workoutlog/configuration/WebSecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+//    authorizeRequests().antMatchers("/resources/**").permitAll().anyRequest().permitAll()
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/user-log/**").hasRole("ADMIN")
+                .antMatchers("/login*", "/registration*", "/css/**", "/js/**", "/lib/**", "/jquery/**", "/measurement-log/upload").permitAll()
+                .anyRequest().permitAll()
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/configuration/WebSecurityConfiguration.java:45]: why do you need row 44 if in 45 you permitting everyone to access your application?

File: src/main/java/by/zadziarnouski/workoutlog/controller/ExerciseEditorController.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+@Controller
+@RequestMapping(path = "/exercise-editor")
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/ExerciseEditorController.java:24]: try not to use separators like '-' in url

File: src/main/java/by/zadziarnouski/workoutlog/controller/ExerciseEditorController.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @GetMapping("/create")
+    public String getCreatePageForExercise(Model model) {
+        Exercise exercise = exerciseService.saveOrUpdate(new Exercise());
+        model.addAttribute("exercise", exerciseMapper.toDTO(exercise));
+        return "create-update-exercise";
+    }
+
+    @PostMapping("/create-update")
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/ExerciseEditorController.java:65]: remove '-' from url

File: src/main/java/by/zadziarnouski/workoutlog/controller/MeasurementLogController.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping(path = "/measurement-log")
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/MeasurementLogController.java:30]: remove '-'

File: src/main/java/by/zadziarnouski/workoutlog/controller/MenuController.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    private final UserService userService;
+    private final UserMapper userMapper;
+
+    @Autowired
+    public MenuController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @RequestMapping(method = {RequestMethod.GET, RequestMethod.POST})
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/MenuController.java:27]: remove POST mapping

File: src/main/java/by/zadziarnouski/workoutlog/controller/PersonalProfileController.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @PostMapping("/create-update")
+    public String createUpdateUser(@ModelAttribute User user, Model model) {
+        user.setRole(Role.ROLE_USER);
+        model.addAttribute("user", userService.saveOrUpdate(user));
+        return "result-create-or-update-user";
+    }
+
+    private UserDetails getPrincipal() {
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/PersonalProfileController.java:79]: Move this method to util class and make it return User. Make this new method static

File: src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Controller
+@RequestMapping(path = "/workout-constructor")
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java:28]: remove '-'

File: src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    @GetMapping("/getFormForAddOrUpdateExercise")
+    public String addExerciseInWorkout(Model model) {
+        model.addAttribute("exercises", exerciseService.findAll().stream().filter(exercise -> exercise.getUser().getId().equals(currentUser.getId())).filter(exercise -> exercise.getNumberOfSets() == 0).map(exerciseMapper::toDTO).collect(Collectors.toList()));
+        model.addAttribute("newExercise", exerciseMapper.toDTO(new Exercise()));
+        return "create-update-exercise-in-workout";
+    }
+
+    @PostMapping("/add-or-update-exercise-in-workout")
```
Comment (RomanKashenok) [inline src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java:73]: name more appropriate please ('store' for example or just 'save')

**Review by RomanKashenok — CHANGES_REQUESTED** [review]

File: src/main/java/by/zadziarnouski/workoutlog/configuration/CloudinaryConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Value("${cloud.name}")
+    private String cloud_name;
+    @Value("${api.key}")
+    private String api_key;
+    @Value("${api.secret}")
+    private String api_secret;
+
+    @Bean
+    public Cloudinary ConnectionToCloudinary() {
```
Comment (aimiebell) [inline src/main/java/by/zadziarnouski/workoutlog/configuration/CloudinaryConfiguration.java:22]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: src/main/java/by/zadziarnouski/workoutlog/controller/PersonalProfileController.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+    @GetMapping
+    public String getPersonalProfilePage(HttpSession session, Model model) {
+        User user = (User) session.getAttribute("user");
+        model.addAttribute("user", userMapper.toDTO(user));
+        return "personal-profile";
+    }
+
+    @PostMapping("/update")
+    public String UpdateProfile(@ModelAttribute UserDTO userDTO, HttpSession session) {
```
Comment (aimiebell) [inline src/main/java/by/zadziarnouski/workoutlog/controller/PersonalProfileController.java:41]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+        Optional<ExerciseDTO> exerciseDTO = workout.getExercises().stream()
+                .filter(exercise -> exercise.getId().equals(id))
+                .map(exerciseMapper::toDTO)
+                .findFirst();
+        exerciseDTO.ifPresent(exercise -> model.addAttribute("newExercise", exercise));
+        return "create-update-exercise-in-workout";
+    }
+
+    @GetMapping("/start-training")
+    public String StartTraining(HttpSession session, Model model) {
```
Comment (aimiebell) [inline src/main/java/by/zadziarnouski/workoutlog/controller/WorkoutConstructorController.java:114]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request closed without merging**